src/ZoneServer/Skills/Handlers/Scout/Ardito/Taglio.cs

```diff
@@ -0,0 +1,175 @@
+
+			if (caster is Character casterCharacter)
+			{
+				if (casterCharacter.Gender == Gender.Male)
+					Send.ZC_PLAY_SOUND(casterCharacter, "voice_war_atk_long_cast");
+				else
+					Send.ZC_PLAY_SOUND(casterCharacter, "voice_atk_long_cast_f");
+			}
+
+			var pad = new Pad(PadName.Arditi_Taglio, caster, skill, new Square(caster.Position, caster.Direction, 45, 30));
+			pad.Position = caster.Position;
+			pad.Trigger.LifeTime = TimeSpan.FromSeconds(maxCastingTime);
+			pad.Trigger.MaxActorCount = 10;
+			pad.Trigger.UpdateInterval = TimeSpan.FromMilliseconds(250);
+			pad.Trigger.Subscribe(TriggerType.Update, this.OnTriggerUpdate);
+
+			caster.Map.AddPad(pad);
+		}
+
+		/// <summary>
+		/// Called when the user stops casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public void EndDynamicCast(Skill skill, ICombatEntity caster)
+		{
+			if (caster.IsBuffActive(BuffId.Taglio_Buff))
+				caster.StopBuff(BuffId.Taglio_Buff);
+
+			if (caster is Character casterCharacter)
+			{
+				Send.ZC_NORMAL.UnkDynamicCastEnd(casterCharacter, skill.Id, 2.1f);
+
+				if (casterCharacter.Gender == Gender.Male)
+					Send.ZC_STOP_SOUND(caster, "voice_war_atk_long_cast");
+				else
+					Send.ZC_STOP_SOUND(caster, "voice_atk_long_cast_f");
+			}
+
+			// Taglio: Tenacity
+			if (!caster.Components.Get<AbilityComponent>().IsActive(AbilityId.Arditi19))
+			{
+				var moveSpeed = this.GetBonusMoveSpeed();
+				caster.Properties.Modify(PropertyName.MSPD_BM, -moveSpeed);
+				Send.ZC_MSPD(caster);
+			}
+		}
+
+		/// <summary>
+		/// Called when an actor enters the area of the attack.
+		/// </summary>
+		/// <param name="sender"></param>
+		/// <param name="args"></param>
+		private void OnTriggerUpdate(object sender, PadTriggerArgs args)
+		{
+			var pad = args.Trigger;
+			var caster = args.Creator;
+			var skill = args.Skill;
+
+			var targets = pad.Trigger.GetActors().Cast<ICombatEntity>().ToList();
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				if (!caster.CanAttack(target))
+					continue;
+
+				this.Attack(skill, caster, target);
+			}			
+		}
+
+		/// <summary>
+		/// Attacks the target one time.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="target"></param>
+		private void Attack(Skill skill, ICombatEntity caster, ICombatEntity target)
+		{
+			var skillHitResult = SCR_SkillHit(caster, target, skill);
+
+			target.TakeDamage(skillHitResult.Damage, caster);
+
+			var hit = new HitInfo(caster, target, skill, skillHitResult, TimeSpan.FromMilliseconds(100));
+			Send.ZC_HIT_INFO(caster, target, hit);
+
+			// Ability - Taglio: Remove Knockback
+			var abilityComponent = caster.Components.Get<AbilityComponent>();
+			if (abilityComponent != null && !abilityComponent.IsActive(AbilityId.Arditi8))
+			{
+				var knockBackDistance = 10;
```

Comment:
I still unsure how to deal with knockbacks: I forgot to insert `target.Position = kb.ToPosition;` on the Granata Skill.
Is this enough for now?

Reply:
Unfortunately this doesn't fix the issue entirely, because the calculations will still be off. I have to admit that the whole knock back thing is in desperate need of some improvements though. I'll fix these two cases.